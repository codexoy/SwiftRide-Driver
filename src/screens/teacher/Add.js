import React from "react"
import { View, Text, Button, StyleSheet, ScrollView } from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Notification from "../../components/Notification"
import PageTitle from "../../components/PageTitle"
import NotificationButtons from "./NotificationButtons"

class Add extends React.Component {
	static navigationOptions = () => {
		return {
			title: "add",
			tabBarLabel: strings("tabs.add"),
			tabBarAccessibilityLabel: strings("tabs.add"),
			tabBarTestID: "AddTab"
		}
	}
	render() {
		return <ScrollView style={styles.container} />
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: 30,
		paddingLeft: 30,
		marginTop: 20
	},
	notification: {}
})

export default connect()(Add)
