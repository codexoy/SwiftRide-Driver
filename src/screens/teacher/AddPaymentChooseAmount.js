import React from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableHighlight,
	TouchableOpacity,
	TextInput,
	KeyboardAvoidingView,
	Platform
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	floatButtonOnlyStyle,
	DEFAULT_MESSAGE_TIME,
	fullButton
} from "../../consts"
import { API_ERROR } from "../../reducers/consts"
import { NavigationActions } from "react-navigation"
import SlidingMessage from "../../components/SlidingMessage"
import { fetchOrError } from "../../actions/utils"
import { popLatestError } from "../../actions/utils"

export class AddPaymentChooseAmount extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			amount: "000",
			slidingMessageVisible: false
		}

		this.addPayment = this.addPayment.bind(this)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	addPayment = async () => {
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/add_payment", {
				method: "POST",
				body: JSON.stringify({
					amount: parseInt(this.state.amount),
					student_id: this.props.navigation.getParam("student")
						.student_id
				})
			})
		)
		if (resp) {
			this.setState({ error: "", slidingMessageVisible: true }, () => {
				setTimeout(
					() =>
						this.props.navigation.dispatch(
							NavigationActions.back()
						),
					DEFAULT_MESSAGE_TIME
				)
			})
		}
	}

	changeAmount = amount => {
		this.setState({ amount })
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.error}
					success={strings("teacher.add_payment.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<View style={styles.headerRow}>
					<TouchableOpacity
						onPress={() => {
							this.props.navigation.goBack()
						}}
					>
						<Icon name="arrow-forward" type="material" />
					</TouchableOpacity>
					<PageTitle
						style={styles.title}
						title={strings("teacher.add_payment.title2")}
					/>
				</View>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : "height"}
					keyboardVerticalOffset={Platform.select({
						ios: 62,
						android: 24
					})}
					style={styles.container}
				>
					<View style={styles.amountContainer}>
						<TextInput
							placeholder="000.00"
							value={this.state.amount}
							onChangeText={amount => this.changeAmount(amount)}
							style={styles.amountInput}
							autoFocus={true}
							maxLength={5}
							keyboardType="number-pad"
						/>
					</View>
					<TouchableHighlight
						underlayColor="#ffffff00"
						onPress={this.addPayment}
						style={styles.floatButton}
					>
						<Text style={styles.buttonText}>{strings("done")}</Text>
					</TouchableHighlight>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService,
		errors: state.errors
	}
}
export default connect(mapStateToProps)(AddPaymentChooseAmount)

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	},
	headerRow: {
		flexDirection: "row",
		padding: 0,
		justifyContent: "flex-start",
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	amountInput: {
		fontSize: 80,
		alignItems: "center",
		fontFamily: "Assistant-Light"
	},
	amountContainer: {
		borderBottomWidth: 0,
		backgroundColor: "#f4f4f4",
		padding: 20,
		marginTop: 80
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold",
		fontSize: 20
	},
	floatButton: {
		...fullButton
	}
})
