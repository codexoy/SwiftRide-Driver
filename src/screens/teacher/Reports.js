import React, { Fragment } from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableOpacity,
	Platform,
	ScrollView,
	TouchableHighlight,
	Linking
} from "react-native"
import { connect } from "react-redux"
import { strings, errors } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	ROOT_URL,
	SHORT_API_DATE_FORMAT,
	floatButtonOnlyStyle,
	DISPLAY_SHORT_DATE_FORMAT
} from "../../consts"
import { Icon } from "react-native-elements"
import { fetchOrError, navigateToEZCount } from "../../actions/utils"
import ShadowRect from "../../components/ShadowRect"
import DateTimePicker from "react-native-modal-datetime-picker"
import moment from "moment"
import AlertError from "../../components/AlertError"

export class Reports extends AlertError {
	constructor(props) {
		super(props)
		this.dates = ["since", "until"]
		this.state = {
			since: moment()
				.startOf("month")
				.format(SHORT_API_DATE_FORMAT),
			until: moment()
				.endOf("month")
				.format(SHORT_API_DATE_FORMAT),
			sinceVisible: false,
			untilVisible: false
		}
	}

	_renderExternalReports() {
		const endpoints = {
			payments: "backoffice/reports/payments",
			incomes: "backoffice/reports",
			export: "backoffice/export",
			revenue: "backoffice/reports/revenue"
		}
		return Object.keys(endpoints).map((endpoint, index) => {
			return (
				<TouchableHighlight
					underlayColor="#f8f8f8"
					onPress={() => {
						this.props.dispatch(
							navigateToEZCount(endpoints[endpoint])
						)
					}}
					style={styles.fullWidth}
					key={endpoint}
				>
					<View style={styles.rectInsideView}>
						<Text>
							{strings("settings.report_types." + endpoint)}
						</Text>
					</View>
				</TouchableHighlight>
			)
		})
	}

	_showDateTimePicker = name => this.setState({ [name + "Visible"]: true })

	_hideDateTimePicker = name => this.setState({ [name + "Visible"]: false })

	_handleDatePicked = (name, date) => {
		this._hideDateTimePicker(name)
		this.setState({ [name]: moment(date).format(SHORT_API_DATE_FORMAT) })
	}

	async createReport(report_type) {
		let data = {
			report_type,
			since: this.state.since,
			until: this.state.until
		}
		const resp = await this.props.dispatch(
			fetchOrError(`/teacher/reports`, {
				method: "POST",
				body: JSON.stringify(data)
			})
		)
		if (resp) {
			Linking.openURL(
				ROOT_URL + "/teacher/reports/" + resp.json["data"]["uuid"]
			)
		}
	}

	_renderDatePickers() {
		return this.dates.map((name, index) => {
			return (
				<DateTimePicker
					isVisible={this.state[name + "Visible"]}
					onConfirm={date => {
						this._handleDatePicked(name, date)
					}}
					onCancel={() => {
						this._hideDateTimePicker(name)
					}}
					key={`picker-${name}${index}`}
				/>
			)
		})
	}

	_renderVisibleDates() {
		let displayDates = {}
		return this.dates.map((name, index) => {
			let style = {}
			if (index % 2 == 0) {
				style = styles.leftSide
			}
			if (this.state[name]) {
				displayDates[name] = moment(this.state[name]).format(
					DISPLAY_SHORT_DATE_FORMAT
				)
			}
			return (
				<TouchableOpacity
					onPress={() => {
						this._showDateTimePicker(name)
					}}
					style={style}
					key={`date-${name}${index}`}
				>
					<View style={styles.dateContainer}>
						<Text style={styles.dateTitle}>
							{strings("settings." + name + "_date")}
						</Text>
						<Text style={styles.dateText}>
							{displayDates[name]}
						</Text>
					</View>
				</TouchableOpacity>
			)
		})
	}

	render() {
		return (
			<ScrollView
				keyboardDismissMode="on-drag"
				keyboardShouldPersistTaps="always"
			>
				<View style={styles.container}>
					<PageTitle
						style={styles.title}
						title={strings("settings.reports")}
						leftSide={
							<TouchableOpacity
								onPress={() => {
									this.props.navigation.goBack()
								}}
								style={styles.closeButton}
							>
								<Icon
									name="ios-close"
									type="ionicon"
									size={36}
								/>
							</TouchableOpacity>
						}
					/>
					<ShadowRect style={styles.rect}>
						<TouchableHighlight
							underlayColor="#f8f8f8"
							onPress={() => {
								this.createReport("students")
							}}
							style={styles.fullWidth}
							key={`studentsReport`}
						>
							<View style={styles.rectInsideView}>
								<Text>
									{strings("settings.report_types.students")}
								</Text>
							</View>
						</TouchableHighlight>
						{this._renderExternalReports()}
					</ShadowRect>
					<Text style={styles.rectTitle}>
						{strings("settings.export_lessons_report")}
					</Text>
					<ShadowRect style={styles.rect}>
						<View style={styles.row}>
							{this._renderVisibleDates()}
						</View>
						<TouchableOpacity
							style={styles.button}
							onPress={() => {
								this.createReport("lessons")
							}}
						>
							<View>
								<Text style={styles.buttonText}>
									{strings("settings.product")}
								</Text>
							</View>
						</TouchableOpacity>
					</ShadowRect>
				</View>
				{this._renderDatePickers()}
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	title: {
		marginTop: 4
	},
	closeButton: {
		marginTop: Platform.select({ ios: -6, android: -12 })
	},
	rect: {
		marginTop: 12,
		marginBottom: 24,
		paddingHorizontal: 0,
		paddingVertical: 0
	},
	rectTitle: {
		fontWeight: "bold",
		alignSelf: "flex-start",
		color: "#5c5959"
	},
	rectInsideView: {
		width: "100%",
		borderBottomWidth: 1,
		borderBottomColor: "#f7f7f7",
		paddingVertical: 6,
		paddingHorizontal: 20,
		paddingVertical: 12,
		flexDirection: "row"
	},
	fullWidth: {
		flex: 1,
		width: "100%"
	},
	button: { ...floatButtonOnlyStyle, width: "100%", borderRadius: 0 },
	buttonText: {
		fontWeight: "bold",
		fontSize: 20,
		color: "#fff"
	},
	row: {
		flex: 1,
		flexDirection: "row",
		alignSelf: "center",
		width: "80%"
	},
	dateContainer: {
		padding: 18
	},
	dateTitle: {
		fontWeight: "bold",
		fontSize: 18,
		alignSelf: "flex-start"
	},
	dateText: {
		alignSelf: "flex-start"
	},
	leftSide: {
		flex: 1,
		marginLeft: "auto",
		alignSelf: "flex-start"
	}
})
function mapStateToProps(state) {
	return {
		error: state.error,
		fetchService: state.fetchService
	}
}
export default connect(mapStateToProps)(Reports)
