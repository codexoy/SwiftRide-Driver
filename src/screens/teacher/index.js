import React from "react"
import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import Home from "./Home"
import Notifications from "./Notifications"
import ChooseDate from "./ChooseDate"
import NewLesson from "./NewLesson"
import Students from "./Students"
import NewStudent from "./NewStudent"
import Schedule from "./Schedule"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"

export default createBottomTabNavigator(
	{
		Home: Home,
		Notifications: Notifications,
		Add: {
			screen: createStackNavigator(
				{ ChooseDate: ChooseDate, NewLesson: NewLesson },
				{
					initialRouteName: "ChooseDate",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "add",
				tabBarLabel: strings("tabs.add"),
				tabBarAccessibilityLabel: strings("tabs.add"),
				tabBarTestID: "NewLessonTab"
			}
		},
		Schedule: Schedule,
		Students: {
			screen: createStackNavigator(
				{ Students: Students, NewStudent: NewStudent },
				{
					mode: "modal",
					initialRouteName: "Students",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "students",
				tabBarLabel: strings("tabs.students"),
				tabBarAccessibilityLabel: strings("tabs.students"),
				tabBarTestID: "StudentsTab"
			}
		}
	},
	{
		defaultNavigationOptions: ({ navigation }) => ({
			tabBarIcon: ({ focused, horizontal, tintColor }) => {
				const { routeName } = navigation.state
				let iconName
				let iconType = "material"
				if (routeName === "Home") {
					iconName = "home"
				} else if (routeName === "Notifications") {
					iconName = "notifications"
				} else if (routeName === "Add") {
					iconName = "add-circle"
				} else if (routeName === "Students") {
					iconName = "people"
				} else if (routeName == "Schedule") {
					iconName = "date-range"
				}

				// You can return any component that you like here!
				return (
					<Icon
						name={iconName}
						size={25}
						color={tintColor}
						type={iconType}
					/>
				)
			}
		}),
		tabBarOptions: {
			activeTintColor: "rgb(12,116,244)",
			inactiveTintColor: "black",
			showLabel: false
		}
	}
)
