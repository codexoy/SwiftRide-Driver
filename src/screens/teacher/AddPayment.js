import { createStackNavigator } from "react-navigation"
import AddPaymentChooseStudent from "./AddPaymentChooseStudent"
import AddPaymentChooseAmount from "./AddPaymentChooseAmount"
export default createStackNavigator(
	{ First: AddPaymentChooseStudent, Second: AddPaymentChooseAmount },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
