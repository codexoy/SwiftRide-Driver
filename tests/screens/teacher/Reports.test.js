import "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { Reports } from "../../../src/screens/teacher/Reports"

describe("Reports", () => {
	test.skip("view renders correctly", () => {
		const tree = renderer.create(<Reports />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
