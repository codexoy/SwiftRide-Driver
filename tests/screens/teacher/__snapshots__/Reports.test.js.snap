// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports view renders correctly 1`] = `
<RCTScrollView
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps="always"
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 26,
          "marginRight": 26,
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 20,
              "marginTop": 4,
            }
          }
        >
          הפקת דוחות
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "marginLeft": "auto",
              "marginTop": -5,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginTop": -6,
                "opacity": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 36,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "elevation": 6,
            "marginBottom": 24,
            "marginTop": 12,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.09,
            "shadowRadius": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text>
              ספר תלמידים
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text>
              תקבולים ותשלומים
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text>
              הכנסות
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text>
              ייצוא קובץ אחיד למס הכנסה
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#f7f7f7",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text>
              רווח והפסד
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#5c5959",
            "fontWeight": "bold",
          }
        }
      >
        הפקת יומן הזמנות
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "elevation": 6,
            "marginBottom": 24,
            "marginTop": 12,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.09,
            "shadowRadius": 20,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "width": "80%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "flex": 1,
                "marginLeft": "auto",
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 18,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                תאריך התחלה
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                  }
                }
              >
                01.05.2019
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "padding": 18,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                תאריך סיום
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                  }
                }
              >
                31.05.2019
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(12,116,244)",
              "borderRadius": 0,
              "elevation": 8,
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "rgb(12,116,244)",
              "shadowOffset": Object {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 16,
              "width": "100%",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              הפק
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <View>
              <RCTDatePicker
                date={1558881845555}
                minuteInterval={1}
                mode="date"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 216,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#d5d5d5",
                "borderTopWidth": 0.5,
                "height": 57,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "height": 57,
              "justifyContent": "center",
              "marginBottom": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <View>
              <RCTDatePicker
                date={1558881845555}
                minuteInterval={1}
                mode="date"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 216,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#d5d5d5",
                "borderTopWidth": 0.5,
                "height": 57,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "height": 57,
              "justifyContent": "center",
              "marginBottom": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
